src/connectors/argentMobile/modal/starknet/adapter.ts

```diff
@@ -84,7 +84,7 @@ export class StarknetAdapter
 
     this.remoteSigner = new StarknetRemoteSigner(this.walletRpc)
 
-    this.provider = new RpcProvider({ nodeUrl: rpcUrl })
+    this.provider = provider ? provider : new RpcProvider({ nodeUrl: rpcUrl })
```

Comment:
@vladutjs let's have a quick chat about this
I was thinking to not pass the provider but only the rpcUrl here (for custom providers)